{%extends 'rolls/layoutpagine.html' %}
{% load static %}

{%block content%}
    <h2 style="color:navy">PANDA Documentation</h2>
   
    <div class="container my-5" style="position: center;">
                <div class="accordion accordion-flush" id="accordionFlushExample">
                    <div class="accordion-item">
                        <h2 class="accordion-header" id="flush-headingOne">
                          <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseOne" aria-expanded="false" aria-controls="flush-collapseOne">
                            DESeq2 analisys
                          </button>
                        </h2>
                        <div id="flush-collapseOne" class="accordion-collapse collapse" aria-labelledby="flush-headingOne" data-bs-parent="#accordionFlushExample">
                          <div class="accordion-body">
                            <p>
                              The differential expression analysis performed with <a href="http://www.bioconductor.org/packages/release/bioc/html/DESeq2.html", target="_blank">DESeq2</a> uses data from the <em>TCGA Gene Expression Quantification HTSeq - counts</em>. 
                              This analysis is based on the <strong>negative binomial distribution</strong> to identify differentially expressed genes.
                          </p>
                          
                          <p>
                              All DESeq2 analyses provided by this web server are precomputed, ensuring that users do not need to spend time on computationally intensive tasks. 
                              This significantly reduces the time required to obtain results.
                          </p>
                          
                          <p>To perform the analysis,</p>
                          
                          <ol>
                            <li><i class="fa-solid fa-arrow-right"></i> Select the tumor of interest</li>
                            <li><i class="fa-solid fa-arrow-right"></i> Press the search button "Features"</li>
                            <li><i class="fa-solid fa-arrow-right"></i> This will display in the second select all the features for which it is possible to obtain the result of this analysis for that specific tumor.</li>
                            <li><i class="fa-solid fa-arrow-right"></i> Pressing "Submit" will display the results with graphs and related information of the data entered by the user.</li>
                          </ol>
                          
                          <p>
                              A key detail provided in the results is the comparison of parameters, specifically the order in which the feature parameters are compared (e.g., <em>Tumor vs. Control</em>).
                          </p>
                          
                          <h5>Results and Output</h5>
                          
                          <p>
                              The analysis results in a downloadable table that includes:
                          </p>
                          <ul>
                              <li>Log2 Fold Change: The calculated fold change between conditions.</li>
                              <li>p-value: The statistical significance of the differential expression.</li>
                              <li>Adjusted p-value (padj):The p-value adjusted for multiple testing to control the false discovery rate.</li>
                          </ul>
                          
                          <p>
                              Additionally, several graphical outputs are available to enhance interpretation, including:
                          </p>
                          
                          <ul>
                              <li>Enhanced Volcano Plot:A plot that highlights significantly differentially expressed genes.</li>
                              <li>Heatmap: A visual representation of gene expression levels across samples.</li>
                              <li>PCA Plot: A Principal Component Analysis plot to visualize sample clustering.</li>
                              <li>Top 100 Heatmap: A heatmap showing the top 100 differentially expressed genes.</li>
                          </ul>
                          
                          <p>
                              These plots can be both viewed directly on the page and downloaded for further analysis or presentation purposes.
                          </p>
                          </div>
                        </div>
                      </div>
                    
                    
                    <div class="accordion-item">
                      <h2 class="accordion-header" id="flush-headingTwo">
                        <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwo" aria-expanded="false" aria-controls="flush-collapseTwo">
                        Differential expression analysis
                        </button>
                      </h2>
                      <div id="flush-collapseTwo" class="accordion-collapse collapse" aria-labelledby="flush-headingTwo" data-bs-parent="#accordionFlushExample">
                        <div class="accordion-body">
                            <h5>Single-gene differential expression analysis</h5>

                            <p>This analysis allows users to explore differences in the expression of a gene, miRNA, or protein between patients in two distinct conditions (based on the selected feature) within a given tumor type. The analysis leverages TCGA Gene Expression Quantification data (FPKM, normalized counts) to assess these differences.</p>
                            <p> The results are visualized in a box plot, where the x-axis represents the two conditions under comparison (e.g., male vs. female), and the y-axis displays the expression levels of the specified gene, miRNA, or protein.</p>
                            <p> A p-value, calculated using the Wilcoxon test, quantifies the significance of the observed expression differences between the two groups.</p>
      
                        </div>
                        </div>
                    </div>
                    

                    <div class="accordion-item">
                      <h2 class="accordion-header" id="flush-headingSix">
                        <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseSix" aria-expanded="false" aria-controls="flush-collapseThree">
                        Differential expression analysis all tumor
                        </button>
                      </h2>
                      <div id="flush-collapseSix" class="accordion-collapse collapse" aria-labelledby="flush-headingSix" data-bs-parent="#accordionFlushExample">
                        <div class="accordion-body">
                            <h5>Differential expression analysis all tumor</h5>
                            <p>This analysis is the differential expression under two conditions of a gene, miRNA, or protein for all tumors for which data for the selected feature is available. 
                                <br>This gives the user an overview of the expression of that particular gene in different tumors.
                                <br>On the x-axis are shown all tumors for which analysis is available. On the y-axis the expression of the gene, miRNA or protein is shown.
                              </p></div>
                      </div>
                    </div>

                    <div class="accordion-item">
                      <h2 class="accordion-header" id="flush-headingThree">
                        <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseThree" aria-expanded="false" aria-controls="flush-collapseThree">
                          miRNA isoform interactor analysis
                        </button>
                      </h2>
                      <div id="flush-collapseThree" class="accordion-collapse collapse" aria-labelledby="flush-headingThree" data-bs-parent="#accordionFlushExample">
                        <div class="accordion-body">
                            <h5>miRNA isoform interactor analysis</h5>
                            <p>This analysis provides the ability for the user to analyze the expression profile of the interactors of miRNA isoforms. 
                              <br>The implemented script checks whether that provided miRNA has isoforms that are present in the RNA inter interaction file. This file contains the information of all miRNA (isoforms) - mRNA interactions
                              that have been experimentally validated belonging to Homo sapiens. 
                              </p>
                              <br>
                              The result obtained is a cluster map. On the rows the names of the patients and on the columns the name of the genes interacting with the miRNA isoform.
                              Clustering was performed to highlight groups of genes with similar expression profile.
                        </div>
                      </div>
                    </div>
                    <div class="accordion-item">
                        <h2 class="accordion-header" id="flush-headingFour">
                          <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseFour" aria-expanded="false" aria-controls="flush-collapseFour">
                            Overall Survival
                          </button>
                        </h2>
                        <div id="flush-collapseFour" class="accordion-collapse collapse" aria-labelledby="flush-headingFour" data-bs-parent="#accordionFlushExample">
                          <div class="accordion-body">
                            <h4>Analysis of overall survival</h4>
                            <p>The analysis presented involves the investigation of overall survival (OS) data using Kaplan-Meier survival curves and log-rank tests to evaluate the association between gene, miRNA, and protein expression levels with patient outcomes. Specifically, OS data provided by The Cancer Genome Atlas (TCGA) is used as time-to-event information. The survival time, expressed in days, measures the duration from diagnosis or treatment initiation until the event of interest, such as death or last follow-up.</p>
                            <p>The expression data used are Gene Expression Quantification HTSeq - FPKM, miRNA Expression Quantification and Protein Expression Quantification</p>
                            <p><b>Choice of Time Variables:</b>Users have the flexibility to select the time variable for the analysis based on their specific research needs. The two available options are:</p>
                                <ol>
                                    <li><strong>OS.time:</strong> This variable uses overall survival time data as the primary measure for the time-to-event analysis.</li>
                                    <li><strong>DFI.time:</strong> This variable utilizes disease-free interval time data, which reflects the duration from diagnosis to the first occurrence of disease recurrence or progression.</li>
                                </ol>
                                <p>The choice between these two time variables can impact the survival analysis results and their interpretations, allowing users to tailor the analysis to their research objectives.</p>

                            <p>The analysis consists of several key components:</p>
                            <p><b>1. Kaplan-Meier Survival Curves:</b> Kaplan-Meier estimators are used to compute the survival probabilities over time for patients stratified by their expression levels. For each gene, miRNA, or protein, patients are divided into two groups: those with expression levels above the median (higher expression) and those below the median (lower expression). The survival curves for each group are plotted to visualize differences in survival probabilities over time.</p>
                        
                            <p><b>2. Log-Rank Test:</b> To statistically compare the survival distributions between the higher and lower expression groups, the log-rank test is employed. This test assesses whether there are significant differences in overall survival between the two groups, with the resulting p-value indicating the level of statistical significance.</p>
                            <p><b>3. Survival Time in Days:</b> The x-axis of the Kaplan-Meier plot represents the time in days, reflecting the OS data. The y-axis displays the survival probability, which represents the conditional probability of surviving beyond a given time point.</p>
                            <p><b>4. Expression Level Impact on Survival:</b> By linking the OS data to gene, miRNA, and protein expression levels, this analysis aims to identify potential biomarkers of survival, providing insights into the biological mechanisms driving cancer progression and patient prognosis.</p>

                            <br>
                            <p>This integrative approach combining transcriptomic, proteomic, and survival data allows for a comprehensive assessment of how molecular alterations impact patient outcomes, potentially guiding personalized cancer therapies and improving prognostic predictions.</p>

                          </div>
                        </div>
                      </div>
                      <div class="accordion-item">
                        <h2 class="accordion-header" id="flush-headingFive">
                          <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseFive" aria-expanded="false" aria-controls="flush-collapseFive">
                            Overall survival with Pathway activity score
                          </button>
                        </h2>
                        <div id="flush-collapseFive" class="accordion-collapse collapse" aria-labelledby="flush-headingFive" data-bs-parent="#accordionFlushExample">
                          <div class="accordion-body">
                            <h4>Overall survival with Pathway activity score</h4>
                            <p>This analysis offers the opportunity to perform an overall survival analysis by going to study instead of the expression of a single gene, the expression of a set of genes (pathways).
                            <br><br>The analysis for a given tumor consists of dividing samples into two groups based on pathway activity score (PAS) values pre-calculated with <a href='https://bioconductor.org/packages/devel/bioc/vignettes/GSVA/inst/doc/GSVA.html' target="_blank">GSVA</a> (Gene Set Variation Analysis).
                            The PAS describes how active that pathway is in that given sample. GSVA is a nonparametric, unsupervised method for estimating the variation in gene set enrichment across samples in an expression dataset.
                            <br>The implemented script uses OS time data provided by TCGA in the clinical data file as time-to-event information. The time data are expressed in days.
                            </p>
                          </div>
                        </div>
                      </div>

                      <div class="accordion-item">
                        <h2 class="accordion-header" id="flush-headingSeven">
                          <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseSeven" aria-expanded="false" aria-controls="flush-collapsSeven">
                            Survival analysis with gene mutation status
                          </button>
                        </h2>
                        <div id="flush-collapseSeven" class="accordion-collapse collapse" aria-labelledby="flush-headingSeven" data-bs-parent="#accordionFlushExample">
                          <div class="accordion-body">
                            <h5>Survival analysis with gene mutation status</h5>
                            <p>This analysis examines the overall survival (OS) of patients by evaluating the impact of specific gene mutations on survival outcomes within a particular tumor type. Utilizing the maftools R package, the analysis leverages mutation data from The Cancer Genome Atlas (TCGA) to provide insights into how the mutation status of a selected gene influences patient prognosis.</p>
     
                            <p></p>
                            <br>
                          </div>
                        </div>
                      </div>


                      <div class="accordion-item">
                        <h2 class="accordion-header" id="flush-heading8">
                          <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapse8" aria-expanded="false" aria-controls="flush-collapse8">
                            Deconvolution
                          </button>
                        </h2>
                        <div id="flush-collapse8" class="accordion-collapse collapse" aria-labelledby="flush-headingFive" data-bs-parent="#accordionFlushExample">
                          <div class="accordion-body">
                           <h5>Deconvolution</h5>
                            <p>

                            </p>
                            <br>
                          </div>
                        </div>
                      </div>
                  </div>
            </div>
            
    </div>
  

   
{%endblock%}
 